### PR TITLE
Add dependency lint

### PR DESCRIPTION
**Pull request by alexdavid**
@kevgo 

There were a lot of dependencies in the wrong place.

This PR adds dependency lint and fixes all the issues

### CODE REVIEW

File: dependency-lint.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# This is the starting point, devFilePatterns and ignoreFilePatterns should be
+# subsets of this pattern
+#   (see https://github.com/isaacs/minimatch)
+filePattern: '**/*.ls'
+
+# array of path patterns to match against a filename
+# to determine if it should be ignored
+#   (see https://github.com/isaacs/minimatch)
+ignoreFilePatterns:
+  - 'node_modules/**/*'
```
Comment (kevgo) [inline dependency-lint.yml]: Btw why isn't dependency-lint smart enough to ignore `node_modules` on its own?

**Comment by kevgo** [comment]
Yeah, this repo went through some wild and intense development! :ship:

File: dependency-lint.yml (hunk 1 above)
Comment (alexdavid) [inline dependency-lint.yml]: This line is generated by dependency-lint, so it is smart enough to know, but I can't think of any reason why someone would want to remove this line, so maybe it should just automatically ignore it.

@charlierudolph what do you think?

Comment (charlierudolph) [inline dependency-lint.yml]: Mainly did this just to make it very explicit what is ignored. If its not there, then I would just be adding a comment to the explantation explaining that `node_modules` is always ignored.

Comment (kevgo) [inline dependency-lint.yml]: For what its worth, `.npmignore` by default ignores the `node_modules` folder. It is enough that this is mentioned in the documentation, there is no need to add a comment about this into each `.npmignore` file. 

I think a similar case could be made here. Not linting external dependencies in node_modules is so obviously the right thing to do that we could simply make it a convention here. If somebody isn't sure and adds node_modules here, we could display a warning. 

I simply see no case where node_modules would ever not be here, at least in correctly set up projects.

**Comment by kevgo** [comment]
:ship:

**Pull request closed without merging**